src/Model/Casts/DateTime/BaseDateTimeCast.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Mongolid\Model\Casts\DateTime;
+
+use DateTimeInterface;
+use MongoDB\BSON\UTCDateTime;
+use MongoDB\BSON\UTCDateTimeInterface;
+use Mongolid\Model\Casts\CastInterface;
+
+abstract class BaseDateTimeCast implements CastInterface
+{
+    /**
+     * @param UTCDateTime|null $value
+     */
+    abstract public function get(mixed $value): DateTimeInterface|null;
```

Comment:
Precisa manter nesse formato de múltiplos tipos?
Sendo um tipo nulo, não dá pra simplificar com interrogação?
```suggestion
     * @param ?UTCDateTime $value
     */
    abstract public function get(mixed $value): ?DateTimeInterface;
```

Reply:
Dá, mas na prática tanto faz.